Add inlay hints documentation (#3269)
rust-analyzer/rust-analyzer: docs/user/features.md

Experimental feature that, given code highlighting using rust-analyzer is
active, will pick unique colors for identifiers.

### Code hints

> Rust Analyzer has two types of hints to show the information about the code:

Reviewer: ```suggestion
Rust-analyzer has two types of hints to show the information about the code:
```

* hover hints, appearing on hover on any element.

> Contains extended information on the language item hovered.

Reviewer: ```suggestion
These contain extended information on the hovered language item.
```

* inlay hints, shown near the element hinted directly in the editor.

Two types of inlay hints are displayed currently:

* type hints, displaying the minimal information on the type of the expression (if the information is available)
* parameter name hints, displaying the names of the parameters in the corresponding methods

#### VS Code

In VS Code, the following settings can be used to configure the inlay hints:

> * rust-analyzer.displayInlayHints — toggles inlay hints display on or off

Reviewer 2: ```suggestion
* `rust-analyzer.displayInlayHints` — toggles inlay hints display on or off
```

> * rust-analyzer.maxInlayHintLength — shortens the hints if their length exceed the value specified. If no value is specified (`null`), no shorteninng is applied.

Reviewer: ```suggestion
* rust-analyzer.maxInlayHintLength — shortens the hints if their length exceeds the value specified. If no value is specified (`null`), no shorteninng is applied.
```

> * rust-analyzer.maxInlayHintLength — shortens the hints if their length exceed the value specified. If no value is specified (`null`), no shorteninng is applied.

Reviewer 2: ```suggestion
* `rust-analyzer.maxInlayHintLength` — shortens the hints if their length exceed the value specified. If no value is specified (`null`), no shortening is applied.
```

> **Note:** VS Code does not have a native support for inlay hints [yet](https://github.com/microsoft/vscode/issues/16221) and the hints are implemented using decorations.

Reviewer: ```suggestion
**Note:** VS Code does not have native support for inlay hints [yet](https://github.com/microsoft/vscode/issues/16221) and the hints are implemented using decorations.
```
